### DISCUSSION TITLE
Hello! and how do I edit the English?

### DISCUSSION THREAD

**Discussion by Janeishly** [david-allison/manx-search-data#1209] [General]
Hello! I'm Jane, born on the IOM, living in Sweden, and trying to learn Manx. I just discovered this corpus today and I'm particularly enjoying the Coraa ny Gael entries as a lot of it's fairly simple and I can just about make sense of some of it. But I've noticed several typos in the English text – I'm a professional editor/proofreader as well as being a Swedish and French to English translator, so noticing things like "between our tow nations" is second nature. It'd take a few seconds to correct – but how do I do it? I think I can see how I'd edit the Manx, but I can't spot mistakes in that!

Although I've often looked stuff up that turns out to be on github, and sometimes even downloaded things, this is the first time I've ever been even remotely tempted to create an account and contribute to anything, so apologies in advance for asking daft questions!

Jane

**Comment by david-allison** [comment]
Hi Jane, thanks for taking an interest!

GitHub's primarily designed as a collaboration platform for software developers. It effectively acts as 'track changes' in Word, but designed for a folder of text files, rather than a single file, with potentially thousands of people editing at once, so the workflows are based around this. 

The corpus could do something more streamlined, but this is an unpaid/hobby project for me, and I've already got more projects that I want to work with than free time, sadly I'm optimizing for 'easy to work with', 'low maintenance' and 'cheap'

To start: The first time you try something technical is always the hardest. GitHub's designed so all mistakes are reversible, so although things are complicated, act with confidence, and they'll become second-nature :)

----

### Explanation

* `david-allison/manx-search-data` is synced with https://corpus.gaelg.im every day

To make a change on GitHub:

* GitHub makes a copy of `david-allison/manx-search-data` under your account: `Janeishly/manx-search-data`
  * You're free to make changes here as you see fit 
* Once you're happy with the changes, you propose that they're synced back to `david-allison/manx-search-data`
* A few automated checks are made to make sure things won't break, I approve things and take responsibility for the changes, and they go live the next day

**Corpus format**

Each file on the corpus is a table. (CSV format)
One column for English, and one column for Manx.
We do most of our editing in Excel, but any text editor will work

The explanation below uses GitHub's online text editor to reduce complexity, once you're familiar with the workflow, we can go through using something a little more user-friendly.

### What to do

* Find a document on the corpus which you wish to edit
  * https://corpus.gaelg.im/docs/Coraa-ny-Gael-18-08-53 
  * Press 'edit'. This takes you to GitHub
  * https://github.com/david-allison/manx-search-data/blob/master/OpenData%2fsecular%20printed%20material%20-%201921%20-%201990%2fCoraa%20ny%20Gael%2fCoraa%20ny%20Gael%201953-08-18/document.csv#L5
* In the top right, press `Fork this repository and edit this file`
* Confirm

<details>

<summary>Click for images</summary>

Image 1): **Fork this repository and edit this file**
<img width="1825" alt="Screenshot 2024-05-03 at 01 43 16" src="https://github.com/david-allison/manx-search-data/assets/62114487/e35ac7ce-3e39-4f20-8976-a48ab636a9aa">

Image 2: **Fork this repository confirmation**
<img width="835" alt="Screenshot 2024-05-03 at 01 43 47" src="https://github.com/david-allison/manx-search-data/assets/62114487/90af1c69-9266-422c-9783-e9686e314429">

</details>

* You'll be able to make changes in the online editor, Ctrl+F to find what you're after, make the edits, then press "commit changes"

<details>

<summary>Click for images</summary>


<img width="1835" alt="Screenshot 2024-05-03 at 01 46 23" src="https://github.com/david-allison/manx-search-data/assets/62114487/b6c8cb7c-c698-4c24-9e4a-953c5b14cdfc">

</details>

Give it a very brief title (`Coraa ny Gael: fix typos`), don't bother with a message

<details>

<summary>Click for images</summary>


<img width="479" alt="Screenshot 2024-05-03 at 01 47 30" src="https://github.com/david-allison/manx-search-data/assets/62114487/39b23d09-8b8d-4b7d-b00d-6c88fabb9424">

</details>

* click "Propose Changes"
* You will see a preview of your changes

<details>

<summary>Click for images</summary>


<img width="1840" alt="Screenshot 2024-05-03 at 01 58 42" src="https://github.com/david-allison/manx-search-data/assets/62114487/b1368521-5b74-49c3-aee3-662b87a1a3ca">

</details>

* Press "Create Pull Request"
* Press "Create Pull Request" again

<details>

<summary>Click for images</summary>


<img width="1840" alt="Screenshot 2024-05-03 at 01 59 46" src="https://github.com/david-allison/manx-search-data/assets/62114487/da15f417-457c-4346-9c92-eaae1133a05a">

</details>


### Terminology

**Fork**: your copy of the text, GitHub lets you make changes to this
**Commit** Changes to 1 or more files
* For now, just treat it as a collection of files to a single file
**Pull Request**: propose that a commit which you made to your fork is applied to `david-allison/manx-search-data` 

----


Feel free to email me using the email on the front page of the Corpus site and we can set up a chat to go through things if there's any questions.

Best wishes, and truly, THANK YOU for taking an interest

**Reply by Janeishly** [reply]
Lovely, thanks - that seems to have worked fine. And thank you for all your hard work!

**Reply by david-allison** [reply]
That was fast! It'll be uploaded tonight. Cheers!

In future, feel free to give me a ping (`@david-allison`) or send an email if things aren't merged within 24 hours

**Reply by Janeishly** [reply]
Oh that was just one article to see how it worked. I'll do more as I read further.
